src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/cargo/ItemFilter.java

```diff
@@ -0,0 +1,175 @@
+package io.github.thebusybiscuit.slimefun4.core.networks.cargo;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+import javax.annotation.Nonnull;
+
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.inventory.ItemStack;
+
+import io.github.thebusybiscuit.slimefun4.utils.SlimefunUtils;
+import io.github.thebusybiscuit.slimefun4.utils.itemstack.ItemStackWrapper;
+import me.mrCookieSlime.CSCoreLibPlugin.Configuration.Config;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+import me.mrCookieSlime.Slimefun.api.inventory.BlockMenu;
+
+/**
+ * The {@link ItemFilter} is a performance-optimization for our {@link CargoNet}.
+ * It is a snapshot of a cargo node's configuration.
+ * 
+ * @author TheBusyBiscuit
+ *
+ */
+class ItemFilter implements Predicate<ItemStack> {
+
+    /**
+     * Our {@link List} of items to check against, might be empty.
+     */
+    private final List<ItemStackWrapper> items = new ArrayList<>();
+
+    /**
+     * Our default value for this {@link ItemFilter}.
+     * A default value of {@literal true} will mean that it returns true if no
+     * match was found. It will deny any items that match.
+     * A default value of {@literal false} means that it will return false if no
+     * match was found. Only items that match will make it past this {@link ItemFilter}.
+     */
+    private boolean defaultValue;
```

Comment:
I don't really like the name of this variable. `rejectItemIfMatched` or something would be a bit clearer.